node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java

```diff
@@ -90,6 +89,27 @@ private void activateNodes(ApplicationId application, Collection<HostSpec> hosts
         nodeRepository.deactivate(activeToRemove, transaction);
         nodeRepository.activate(updateFrom(hosts, continuedActive), transaction); // update active with any changes
         nodeRepository.activate(updatePortsFrom(hosts, reservedToActivate), transaction);
+        unreserveParentsOf(reservedToActivate);
+    }
+
+    /** When a tenant node is activated on a host, we can open up that host for use by others */
+    private void unreserveParentsOf(List<Node> nodes) {
+        List<String> reservedParents = nodes.stream()
+                                            .filter(node -> node.parentHostname().isPresent())
+                                            .map(node -> nodeRepository.getNode(node.parentHostname().get()))
+                                            .filter(parent -> parent.isPresent())
+                                            .filter(parent -> parent.get().reservedTo().isPresent())
+                                            .map(parent -> parent.get().hostname())
+                                            .collect(Collectors.toList());
+        if (reservedParents.isEmpty()) return;
+
+        try (Mutex lock = nodeRepository.lockUnallocated()) {
```

Comment:
This should lock the tenant host application

Reply:
That stuff has become pretty weird, and it doesn't seem that's how it is consistently done? Anyway; ok, done.